### PR TITLE
feat: snapshot voting with markdown editor

### PR DESCRIPTION
**Pull request by mindsgn**
## What changes

- added a snapshot proposal form to enable voting via the website.

## Anything to note for reviewers?

-

## Screenshots (if applicable)

Dec-30-2022 13-36-35

## Issues

-

## Resolves

- Resolves [Add link to issue here]

### CODE REVIEW

**Review by niikkiin — CHANGES_REQUESTED** [review]
Hey @mindsgn! Thanks for contributing! Did a quick run through on the dapp for your PR.

- The styling on the view snapshots may have been affected on the recent changes. Was the `react-markdown.css` removed from the imports?
image

- The end date should be set automatically and are disabled by default like in snapshot. I think it would make more sense if the past dates are disabled.
image
image
- Would it be possible to add a height limit on the content?
- It would also be helpful if we have a title before the start date and end dates. Please add "Voting Period" and the tooltip that says the same thing here.
image
- Can you also widen the modal a bit? About 50% of the screen would be nice.
- Please disable the submit button when fields are empty.
- I'm not able to create a snapshot due to this error. Not sure which went wrong here. I just copy and pasted a markdown from previous proposal. @arjanvillon said it's already opened in public, we can discuss this on discord.
image
- When clicking a proposal, it's refreshing which does not happen before. Looked at the network and I think it might be because the post snapshot might be running when navigating

https://user-images.githubusercontent.com/26428605/210227193-d12f7bfe-cf94-419d-ba91-2485125b8dd5.mov

File: src/components/Forms/ProposalForm.tsx [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+      end: Yup.number()
+        .min(
+          Date.parse(`${new Date()}`) - 1000000,
+          `Date must be greater than ${new Date()}`
+        )
+        .required('Required'),
+      choices: Yup.array().min(2, `2 choices or more must be provided`),
+    }),
+    onSubmit: async (values: Proposal) => {
+      console.log(values)
```
Comment (niikkiin) [inline src/components/Forms/ProposalForm.tsx:78]: let's remove console log here. thanks!

File: src/components/Forms/ProposalForm.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+              variant: 'error',
+            })
+
+            return null
+          }
+
+          await client.proposal(web3, account, values)
+
+          toast({
+            title: 'Proposal Failed',
```
Comment (niikkiin) [inline src/components/Forms/ProposalForm.tsx:124]: I'm assuming this should be `Proposal Created` right?

File: src/components/Forms/ProposalForm.tsx [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    },
+  })
+
+  const [items, setItems] = React.useState<string[]>(formik.values.choices)
+
+  const addItem = (item: string) => {
+    item = item.trim()
+    if (item === '' || items.includes(item)) {
+      toast({
+        title: 'choice not added',
```
Comment (niikkiin) [inline src/components/Forms/ProposalForm.tsx:153]: Please capitalize the first letter here. `Choice not added.` thank you!

File: src/components/Forms/ProposalForm.tsx [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+        description: 'choice input is empty or it exists',
```
Comment (niikkiin) [inline src/components/Forms/ProposalForm.tsx:154]: `Choice input is empty or it exists already.`

File: src/pages/_app.tsx [hunk 5]
```diff
@@ -19,7 +21,6 @@ import { VotingProvider } from 'store/contexts/votingContext'
 import { VeNewoProvider } from 'store/contexts/veNewoContext'
 
 import 'styles/date-range-picker.css'
-import 'styles/react-markdown.css'
```
Comment (niikkiin) [inline src/pages/_app.tsx:22]: This is the styling for the view snapshot pages.

File: package.json [hunk 6]
```diff
@@ -22,14 +22,17 @@
     "@emotion/styled": "^11",
     "@metamask/onboarding": "^1.0.1",
     "@snapshot-labs/snapshot.js": "^0.4.5",
+    "@uiw/react-markdown-editor": "^5.10.1",
```
Comment (niikkiin) [inline package.json:25]: Is this package used somewhere in the form?

File: package.json [hunk 7]
```diff
@@ -39,15 +42,20 @@
     "react": "18.1.0",
     "react-datepicker": "^4.8.0",
     "react-dom": "18.1.0",
+    "react-draft-wysiwyg": "^1.15.0",
     "react-error-boundary": "^3.1.4",
     "react-hook-form": "^7.33.0",
     "react-icons": "^4.4.0",
     "react-markdown": "^8.0.3",
+    "react-markdown-editor-lite": "^1.3.4",
```
Comment (niikkiin) [inline package.json:50]: Was this also used?

File: src/pages/_app.tsx (hunk 5 above)
Comment (mindsgn) [reply src/pages/_app.tsx:22]: fixed

File: .env [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+NEXT_PUBLIC_HUB_URL=https://hub.snapshot.org
+
+NEXT_PUBLIC_VERCEL_ENV=development
+
+NEXT_PUBLIC_UPSTASH_REDIS_REST_URL=redis://default:ad9abd17ad064d92881873a2bd2d74d7@us1-refined-mullet-37898.upstash.io:37898
+NEXT_PUBLIC_UPSTASH_REDIS_REST_TOKEN=ad9abd17ad064d92881873a2bd2d74d7
+
+SECRET_API_KEY=
+
+NEXT_PUBLIC_IMAGE_BASE_URL=https://ipfs.io
```
Comment (niikkiin) [inline .env:20]: could you please remove this env file?

File: package.json (hunk 6 above)
Comment (niikkiin) [reply package.json:25]: can we remove unused packages such as `@uiw/react-markdown-editor` and `react-markdown-editor-lite`?

File: src/pages/_app.tsx (hunk 5 above)
Comment (niikkiin) [reply src/pages/_app.tsx:20]: i'm still not seeing this import on the `_app.tsx` can you please double check? thanks

**Pull request closed without merging**